### ISSUE TITLE
How does SC 2.1.1 apply to long-press?

### ISSUE THREAD

**Issue by mitchellevan** [w3c/matf#275]
Assuming an app function is only available through long press, under what circumstances would it pass or fail SC 2.1.1 Keyboard?

### Currently

WCAG [Understanding SC 2.1.1](https://www.w3.org/WAI/WCAG22/Understanding/keyboard.html) says:

> Examples of "specific timings for individual keystrokes" include situations where a user would be required to repeat or execute multiple keystrokes within a short period of time or where a key must be held down for an extended period before the keystroke is registered.

Currently neither MATF nor WCAG2ICT clarifies whether a keyboard equivalent of longpress fails the "specific timing" requirement of SC 2.1.1.

### Proposed

iOS offers SPACE (press and hold) as a way of achieving a long-press touch, but this is not sufficient to meet SC 2.1.1.
* It works only after TAB+G, which sends touch commands. Since it is not using the keyboard interface, it is technically not a keyboard mechanism.
* It works only after a specific minimum time delay. So even if it were considered a keyboard interface mechanism, apps relying on this mechanism would still fail the "specific timing" requirement for SC 2.1.1.

iOS also offers TAB+M to display a contextual menu. Where this performs the same function as a long press, it would pass SC 2.1.1.

Android offers ENTER (press and hold) as a keyboard equivalent for a long-press touch. Android also lets users configure their [Touch & hold delay](https://mcmw.abilitynet.org.uk/how-to-increase-the-touch-and-hold-delay-in-android-15), which applies equally to touch and keyboard long-press. So it would appear that apps relying on Android's keyboard mechanism would pass SC 2.1.1, including the "specific timing" requirement.

(Google does not document its long-press mechanism for keyboard users, which arguably puts a burden on Android app developers to meet SC 3.3.2 Labels or Instructions, but that's a separate matter.)

Is the above all correct? It would be great to see this clarified in MATF notes for 2.1.1.

**Comment by JJdeGroot** [comment]
@mitchellevan FYI: linked this to our main issue for 2.1.1, which is in scope for our third draft, which we are now starting on
